src/main/java/soot/jimple/toolkits/callgraph/VirtualCalls.java

```diff
@@ -239,13 +240,19 @@ protected void resolveAnySubType(Type declaredType, Type sigType, NumberedString
       }
     }
 
-    Set<Type> newSubTypes = new HashSet<>();
-    newSubTypes.add(base);
+    Set<Type> newSubTypes = new HashSet<>();    
+    
 
     LinkedList<SootClass> worklist = new LinkedList<SootClass>();
     HashSet<SootClass> workset = new HashSet<SootClass>();
     FastHierarchy fh = fastHierachy;
     SootClass cl = base.getSootClass();
+    
+    //Do not add abstract classes and interfaces as they do not contain implementations
+    if(!cl.isAbstract() && !cl.isInterface())
```

Comment:
@mbenz89 Added a condition to check if the class is not abstract and not an interface. After adding the check the number of edge difference got reduced from 429 to 143.

I am thinking we should still be allowing abstract classes to get added as they may contain an implementation because an abstract class can contain non abstract methods as well right?

Reply:
You are right, it depends on if the method is abstract or not. But for this case, please just change it back as in the original develop branch. We want to compare to that state.

Reply:
Okay. So I should not be making any changes to develop but just check why those differences are present right?

Reply:
Exactly. We want to know if there are issues in the pr. We would only fix something if it is not fixed by the pr anyway.